graphql/resolvers/addComment.ts

```diff
@@ -11,6 +11,137 @@ import {
 } from '../../constants'
+import { Challenge, Comment, Lesson, Submission } from '@prisma/client'
+import { MessageResponse } from '../../@types/discordBot'
+
+type CommentWithAuthorAndSubmission = Comment & {
+  author: {
+    id: number
+    username: string
+    discordId: string | null
+    name: string
+  }
+  submission: Submission & {
+    user: { id: number; username: string; discordId: string | null }
+    lesson: Lesson
+    challenge: Challenge
+  }
+}
+
+type SendNotificationsParams = {
+  line: InputMaybe<number> | undefined
+  fileName: InputMaybe<string> | undefined
+  submissionId: number
+  comment: CommentWithAuthorAndSubmission
+}
+
+const sendNotifications = async ({
+  line,
+  submissionId,
+  fileName,
+  comment
+}: SendNotificationsParams) => {
+  const restOfComments = await prisma.comment.findMany({
+    where: {
+      line,
+      submissionId,
+      fileName
+    },
+    select: {
+      author: {
+        select: {
+          id: true,
+          discordId: true
+        }
+      }
+    }
+  })
+
+  const { submission, author } = comment
+
+  const notificationEmbed = (forSubmissionOwner: boolean): APIEmbed => {
+    const params = {
+      title: forSubmissionOwner
+        ? `New comment by a reviewer on submission`
+        : `New comment on submission`,
+      description: forSubmissionOwner
+        ? 'commented on your submission to the'
+        : 'added a comment on'
+    }
+
+    return {
+      color: PRIMARY_COLOR_HEX,
+      title: params.title,
+      url: `${CURRICULUM_URL}/${submission.lesson.slug}`,
+      thumbnail: {
+        url: getLessonCoverPNG(submission.lesson.order)
+      },
+      author: {
+        name: author.name,
+        url: `${PROFILE_URL}/${author.username}`,
+        icon_url: C0D3_ICON_URL
+      },
+      description: `${getDiscordMessageUserIdString(author)} ${
+        params.description
+      } challenge **${submission.challenge.title}**${
+        comment.line ? ` on **Line ${comment.line}**` : ''
+      }.`,
+      fields: [
+        {
+          name: 'Comment',
+          value: comment.content
+        }
+      ]
+    }
+  }
+
+  // sends a notification to the submission author
+  const sendNotificationToAuthor = async () => {
+    if (author.id !== submission.user.id && submission.user.discordId) {
+      await sendDirectMessage(
+        submission.user.discordId,
+        '',
+        notificationEmbed(true)
+      )
+    }
+  }
+
+  // sends a notification to all of those who commented on
+  // the same submission line except the comment author and the student who made the submission
+  const sendNotificationsToParticipants = async () => {
+    const notifications = restOfComments.reduce(
+      (acc: Promise<MessageResponse>[], participant) => {
+        if (
+          participant.author.id !== author.id ||
+          participant.author.id === submission.user.id
+        ) {
+          return acc
+        }
+
+        if (participant.author.discordId) {
```

Comment:
Combine this condition with the if above as well (and change is accordingly). If the above if is true or if this if is false, the same thing is happening (acc is being returned). And this if does not depend on the above check, so you can make this simpler and reduce the number of paths this code can take here.